src/gui/transferlistsortmodel.h

```diff
@@ -64,7 +66,9 @@ class TransferListSortModel final : public QSortFilterProxyModel
 
     TorrentFilter m_filter;
     mutable CachedSettingValue<int> m_subSortColumn;
+    mutable CachedSettingValue<int> m_subSortColumnOrder;
     mutable int m_lastSortColumn = -1;
+    mutable int m_lastSortColumnOrder = 0;
```

Comment:
It looks wasteful to use 4 bytes to store very few items. I would suggest using `signed char` (`signed char` to differentiate with the usual `char` type).
This applies to `CachedSettingValue<>` too.

Reply:
I think it's too much... This is not a memory-critical place.

Reply:
I can use single `int8_t` to encode both column index and sort order at the same time so every user will waste 7 bytes less at the expense of some runtime overhead and inability to express more than 126 columns. Sort order will be encoded using sign bit and index itself will be stored with +1 offset, 0 will mean "unset" which also allows to save on code size by replacing `if (m_lastSortColumn != -1)` with much shorter `if (m_lastSortColumn)`. If we agree to never have more than 62 columns we may even spare 1 bit for future use.

Reply:
>I think it's too much...

It is only a smaller type isn't? Or we can use the alias `qint8`.

>I can use single int8_t to encode both column index and sort order at the same time so every user will waste 7 bytes less at the expense of some runtime overhead and inability to express more than 126 columns. 

That is a way to encode it but it is overly complex for our simple usage here. For now let's stick to simple code.